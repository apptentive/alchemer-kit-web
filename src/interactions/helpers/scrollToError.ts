/**
 * Scrolls errors into view.
 *
 * @param {*} _event - Unused
 * @private
 */
const scrollToError = (_event: Event) => {
  const errors = document.querySelectorAll('.apptentive-survey-questions .invalid');

  if (errors.length > 0) {
    errors[0].scrollIntoView();
  }
};

export default scrollToError;
