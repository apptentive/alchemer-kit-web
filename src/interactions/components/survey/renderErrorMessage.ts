import { surveySelectors } from '../../../constants/elementSelectors';
import scrollToError from '../../helpers/scrollToError';

/**
 * Renders a given surveys error container.
 *
 * @returns {HTMLElement} - The DOM Node
 */
const renderErrorMessage = () => {
  const errors = document.createElement(surveySelectors.errorMessage);
  errors.className = 'apptentive-errors';
  errors.textContent = '';
  errors.addEventListener('click', scrollToError);
  errors.style.display = 'none';

  return errors;
};

export default renderErrorMessage;
