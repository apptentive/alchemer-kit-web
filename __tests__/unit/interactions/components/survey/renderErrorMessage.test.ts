import renderErrorMessage from '../../../../../src/interactions/components/survey/renderErrorMessage';
import scrollToError from '../../../../../src/interactions/helpers/scrollToError';

jest.mock('../../../../../src/interactions/helpers/scrollToError.ts');

describe('renderErrorMessage', () => {
  test('properly renders error message container', () => {
    const output = renderErrorMessage();

    expect(output).toMatchSnapshot();
  });

  test('properly attaches scroll event to container', () => {
    const output = renderErrorMessage();
    document.body.append(output);

    output.click();

    expect(scrollToError).toHaveBeenCalledTimes(1);

    document.body.innerHTML = '';
  });
});
