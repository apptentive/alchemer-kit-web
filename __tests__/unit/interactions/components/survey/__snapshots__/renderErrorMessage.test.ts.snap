// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderErrorMessage properly renders error message container 1`] = `
<apptentive-errors
  class="apptentive-errors"
  style="display: none;"
/>
`;
